src/Symfony/Component/ObjectMapper/Attribute/Map.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Attribute;
+
+/**
+ * Configures a class or a property to map to.
+ *
+ * @experimental
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ *
+ * @psalm-type CallableType = string|callable(mixed $value, object $object): mixed
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+readonly class Map
+{
+    /**
+     * @param string|class-string|null                                      $source    The property or the class to map from
+     * @param string|class-string|null                                      $target    The property or the class to map to
+     * @param string|bool|callable(mixed $value, object $object): bool|null $if        A boolean, Symfony service name or a callable that instructs whether to map
+     * @param CallableType|CallableType[]|null                              $transform A Symfony service name or a callable that transform the value during mapping
```

Comment:
it has been decided above that the introduction of this component in the FrameworkBundle would be done in another PR.

Reply:
Fine with doing the framework integration in another PR, but I wouldn't merge this one without having that PR open as there's chances it raises questions about the component.

Reply:
And I wouldn't release a component without framework integration

Reply:
But if this component is not merged why would one take days to add the framework integration?

Reply:
See [`e9b915c` (#58654)](https://github.com/symfony/symfony/pull/58654/commits/e9b915cb2cdf006bb80b6d190534863f730f3eba)

Reply:
@soyuka because this might allow uncovering cases where the proposed component implementation is incompatible with the way we would like to configure it in a typical framework usage (with cached metadata and so on)